app/lib/features/chat/widgets/custom_input.dart

```diff
@@ -1040,6 +898,126 @@ class _TextInputWidget extends ConsumerWidget {
       chatInputNotifier.emojiPickerVisible(false);
     }
   }
+
+  @override
+  Widget build(BuildContext context) {
+    ref.listen(chatInputProvider(widget.roomId), (prev, next) {
+      if (next.selectedMessageState == SelectedMessageState.edit &&
+          (prev?.selectedMessageState != next.selectedMessageState ||
+              next.message != prev?.message)) {
+        // a new message has been selected to be edited or switched from reply
+        // to edit, force refresh the inner text controller to reflect that
+        widget.controller.text = next.message;
+        chatFocus.requestFocus();
+      } else if (next.selectedMessageState == SelectedMessageState.replyTo &&
+          (next.selectedMessage != prev?.selectedMessage ||
+              prev?.selectedMessageState != next.selectedMessageState)) {
+        chatFocus.requestFocus();
+      }
+    });
+    return CallbackShortcuts(
+      bindings: <ShortcutActivator, VoidCallback>{
+        const SingleActivator(LogicalKeyboardKey.enter): () {
+          widget.onSendButtonPressed();
+        },
+      },
+      child: MultiTriggerAutocomplete(
```

Comment:
only this part actually update, am I right?

Reply:
Yes indeed.